src/MIDI.hpp

```diff
@@ -80,34 +82,33 @@ inline MidiInterface<SerialPort, Settings>::~MidiInterface()
  - Input channel set to 1 if no value is specified
  - Full thru mirroring
  */
-template<class SerialPort, class Settings>
-void MidiInterface<SerialPort, Settings>::begin(Channel inChannel)
+template<class Transport, class Settings, class Platform>
+void MidiInterface<Transport, Settings, Platform>::begin(Channel inChannel)
 {
     // Initialise the Serial port
-#if defined(AVR_CAKE)
-    mSerial. template open<Settings::BaudRate>();
-#else
-    mSerial.begin(Settings::BaudRate);
-#endif
+    mTransport.begin();
 
     mInputChannel = inChannel;
     mRunningStatus_TX = InvalidType;
     mRunningStatus_RX = InvalidType;
 
     mPendingMessageIndex = 0;
-    mPendingMessageExpectedLenght = 0;
 
     mCurrentRpnNumber  = 0xffff;
     mCurrentNrpnNumber = 0xffff;
 
+    mSenderActiveSensingPeriodicity = Settings::SenderActiveSensingPeriodicity;
+    mLastMessageSentTime = Platform::now();
+
     mMessage.valid   = false;
     mMessage.type    = InvalidType;
     mMessage.channel = 0;
     mMessage.data1   = 0;
     mMessage.data2   = 0;
+    mMessage.length  = 0;
 
     mThruFilterMode = Thru::Full;
-    mThruActivated  = true;
+    mThruActivated  = false;
```

Comment:
To minimize behaviour changes, thru should stay active by default.

Reply:
OK